### PR TITLE
ci(DCP-2583): automated release process on skip-release tag

### LINKED ISSUE #358: Automated Release Process
Complete the automated release process for the CLI, picking up where [DCP-2422](https://prolific.atlassian.net/browse/DCP-2422) left off.

## Current State

Conventional commits enforcement and local changelog tooling are in place (PRs #321, #322). A full release automation workflow was built and merged (#322) but had to be rolled back (#340) due to CI permissions constraints.

The release process is currently manual: a developer must run `make changelog`, draft a GitHub Release in the UI, create a tag, and publish to trigger binary builds.

## What's Needed

Automate the release process so that a release can be triggered with minimal manual steps, handling: version calculation from conventional commits, changelog generation via git-cliff, tagging, and GitHub Release creation — triggering existing binary/Docker build workflows.

### Ideal: Single-action release
One workflow dispatch (choosing patch/minor/major) handles everything end-to-end.

### Acceptable: Two-step release
1. **Generate changelog** — run `make changelog VERSION=x.y.z` locally or via CI to update `CHANGELOG.md` and commit
2. **Trigger release workflow** — dispatch a workflow that reads the latest changelog entry, creates a git tag, publishes a GitHub Release, and triggers binary/Docker builds

The two-step approach avoids the CI permissions complexity of creating PRs/commits from workflows while still eliminating the manual GitHub Releases UI work.

## Prior Art

- PR #322 — original automation implementation
- PR #340 — rollback, retaining changelog tooling
- `cliff.toml` and `scripts/changelog/` — already in the codebase and working

[DCP-2422]: https://prolific.atlassian.net/browse/DCP-2422?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### PR DESCRIPTION
**Pull request by dessskris**
This pull request implements a robust, automated workflow for managing releases via GitHub Actions, ensuring consistency and correctness in changelog and release processes. The main changes introduce new workflow files for release gating, automate release creation, and extend the `changelog` script with version extraction logic and tests.

**Release automation and gating:**

* Added `.github/workflows/changelog-gate.yml` to require that any PR with `[run-release]` in the title must include a `CHANGELOG.md` update, preventing accidental releases without changelog entries.
* Added `.github/workflows/release-tag-gate.yml` to validate that PR titles use the exact `[run-release]` tag (case and spelling), catching common mistakes and preventing mis-triggered releases.
* Introduced `.github/workflows/create-release.yml` to automate release creation: when a `[run-release]` PR is merged to `main`, it extracts the version from `CHANGELOG.md`, creates a tag, generates release notes, and triggers the build workflow.

**Release build workflow enhancements:**

* Updated `.github/workflows/release.yml` to accept a release tag as input, check out the correct tag, and use it for building and uploading artifacts, enabling both manual and automated (workflow_call) releases. [[1]](diffhunk://#diff-87db21a973eed4fef5f32b267aa60fcee5cbdf03c67fafdc2a9b553bb0b15f34R6-R31) [[2]](diffhunk://#diff-87db21a973eed4fef5f32b267aa60fcee5cbdf03c67fafdc2a9b553bb0b15f34L49-R65)

**Changelog tooling improvements:**

* Extended `scripts/changelog/main.go` with a new `extract-version` command and `ExtractReleaseVersion` function to reliably extract the first strict semver version from `CHANGELOG.md`, skipping "next" and non-semver headings. [[1]](diffhunk://#diff-34ee22bcfa2978e1501e8dd13f91d168576882a268763a1c8ee80d4bb310b5e4R7) [[2]](diffhunk://#diff-34ee22bcfa2978e1501e8dd13f91d168576882a268763a1c8ee80d4bb310b5e4R29-R34) [[3]](diffhunk://#diff-34ee22bcfa2978e1501e8dd13f91d168576882a268763a1c8ee80d4bb310b5e4R44-R45) [[4]](diffhunk://#diff-34ee22bcfa2978e1501e8dd13f91d168576882a268763a1c8ee80d4bb310b5e4R103-R129) [[5]](diffhunk://#diff-34ee22bcfa2978e1501e8dd13f91d168576882a268763a1c8ee80d4bb310b5e4R228-R248)
* Added comprehensive unit tests for `ExtractReleaseVersion` in `scripts/changelog/main_test.go`, covering edge cases like missing sections, pre-releases, and malformed headings.

Fixes #358

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -3,19 +3,32 @@ name: Release
 on:
   release:
     types: [published]
+  workflow_call:
+    inputs:
+      tag:
+        required: true
+        type: string
```
[the remark this answers is not in the document]
Comment (dessskris) [reply .github/workflows/release.yml]: done
